rocAL/rocAL/source/pipeline/master_graph.cpp

```diff
@@ -793,15 +793,84 @@ MasterGraph::copy_out_tensor(void *out_ptr, RocalTensorFormat format, float mult
+                                mask_G = _mm256_setr_epi32(0x80808001, 0x80808004, 0x80808007, 0x8080800A, 0x80808001,
+                                                            0x80808004, 0x80808007, 0x8080800A);
+                                mask_R = _mm256_setr_epi32(0x80808002, 0x80808005, 0x80808008, 0x8080800B, 0x80808002,
+                                                            0x80808005, 0x80808008, 0x8080800B);
+                            } else {
+                                mask_R = _mm256_setr_epi32(0x80808000, 0x80808003, 0x80808006, 0x80808009, 0x80808000,
+                                                            0x80808003, 0x80808006, 0x80808009);
+                                mask_G = _mm256_setr_epi32(0x80808001, 0x80808004, 0x80808007, 0x8080800A, 0x80808001,
+                                                            0x80808004, 0x80808007, 0x8080800A);
+                                mask_B = _mm256_setr_epi32(0x80808002, 0x80808005, 0x80808008, 0x8080800B, 0x80808002,
+                                                            0x80808005, 0x80808008, 0x8080800B);
+                            }
+                            __m256 pmul0 = _mm256_set1_ps(multiplier0);
+                            __m256 pmul1 = _mm256_set1_ps(multiplier1);
+                            __m256 pmul2 = _mm256_set1_ps(multiplier2);
+                            __m256 padd0 = _mm256_set1_ps(offset0);
+                            __m256 padd1 = _mm256_set1_ps(offset1);
+                            __m256 padd2 = _mm256_set1_ps(offset2);
+                            unsigned int alignedLength = (channel_size & ~7);    // multiple of 8
+                            unsigned int i = 0;
+
+                            __m256 fR, fG, fB;
+                            float tempR[8], tempG[8], tempB[8];
+                            for (; i < alignedLength; i += 8) {
+                                __m256i pix0 = _mm256_loadu_si256((const __m256i *) in_buffer);
+                                pix0 = _mm256_permutevar8x32_epi32(pix0, _mm256_setr_epi32(0, 1, 2, 3, 3, 4, 5, 6));
+                                fB = _mm256_cvtepi32_ps(_mm256_shuffle_epi8(pix0, mask_R));
+                                fG = _mm256_cvtepi32_ps(_mm256_shuffle_epi8(pix0, mask_G));
+                                fR = _mm256_cvtepi32_ps(_mm256_shuffle_epi8(pix0, mask_B));
+                                fB = _mm256_fmadd_ps(fB, pmul0, padd0);
+                                fG = _mm256_fmadd_ps(fG, pmul1, padd1);
+                                fR = _mm256_fmadd_ps(fR, pmul2, padd2);
+                                _mm256_storeu_ps(tempB, fB);
+                                _mm256_storeu_ps(tempG, fG);
+                                _mm256_storeu_ps(tempR, fR);
+                                for(int cnt = 0; cnt < 8; cnt++)
+                                {
+                                    B_buf_16[cnt] = (half) tempB[cnt];
```

Comment:
please do the conversion from float to half in SSE using _mm_cvtps_ph() or similar instruction and get rid of loop here

Reply:
Created a new commit adding _mm_cvtps_ph() for conversion from float to half